### ISSUE TITLE
IgnoreExceptions does not support ignoring assertion errors

### ISSUE THREAD

**Issue by wesamhaboush** [awaitility/awaitility#103]
would like to catch assertion errors, and be able to ignore some of them. If the interface of ignore exceptions takes a throwable, it should really work for all throwables, including errors. No?

**Comment by johanhaleby** [comment]
AssertionErrors are handled in a special way in order to integrate with other libraries such as AssertJ and thus you can't catch them. Why would you need to do this?

**Comment by wesamhaboush** [comment]
@johanhaleby We have a use case where some assertion errors should not bubble up. It caught us off guard when we looked at the code and found that `AssertionError` are treated differently. At the end of the day the Matcher is of `Throwable`, so seems we should be able to do whatever we want. We either can catch all `Throwables` as advertised by the contract of the matcher, or we change the matcher may be to `Exception` matcher. Also we can make that configurable via the build, as in `.doNotTreatAssertionErrorsSpecial().ignoreExceptions(matchXyz);`

**Comment by johanhaleby** [comment]
Hmm maybe awaitility could wrap assertion errors internally so that they can be ignored. But I worry about backward incompatibilities if we do this...

**Comment by wesamhaboush** [comment]
@johanhaleby may be you can make that behavior configurable with the default not to do so to maintain backward compatibility?

**Comment by johanhaleby** [comment]
That would indeed be a possibility, but we also have to wonder if it's worth the extract complexity of introducing this? Maybe it would.

**Comment by wesamhaboush** [comment]
Yep, it is definitely more complex. I am asking myself the following question: is maintaining the fidelity on the matcher for throwable interface worth it?

Seems from an pure engineering perspective, it is a good thing, but back to reality, I would understand it might break things rendering purity an utter BS in the context. 

What about the new major release of awaitability? would you have a window of opportunity to break that backward compatibility without having to justify it?
